modules/core/src/matrix.cpp

```diff
@@ -664,6 +664,8 @@ void Mat::create(int d, const int* _sizes, int _type)
 
     if( data && (d == dims || (d == 1 && dims <= 2)) && _type == type() )
     {
+        if ( d == 1 && _sizes[0] == size[0] )
```

Comment:
@alalek Could you have a look:
below there is a condition:
```
if( i == d && (d > 1 || size[1] == 1))
    return;
```
Checking `size[1]` will cause error, because `d <= 1`.
Is it valid to check `size[1]` in that case?
https://github.com/opencv/opencv/blob/4.x/modules/core/src/matrix.cpp#L672

Reply:
`&& dims == 1` to preserve old behavior for requesting 1D size of pre-allocated 2D.

---

`size[1]` was valid (without 1D hacks). We could add `dims == 2` precheck there.

Reply:
Added